extensions/pkg/controller/worker/machines.go

```diff
@@ -165,6 +166,10 @@ func WorkerPoolHash(pool extensionsv1alpha1.WorkerPool, cluster *extensionscontr
 		}
 	}
 
+	if helper.IsNodeLocalDNSEnabled(cluster.Shoot.Spec.SystemComponents, cluster.Shoot.Annotations) {
```

Comment:
Indeed, this would be the current effect.
Actually, we only need to roll the nodes when disabling node local DNS, but I am not aware how to achieve this with the means we have (`WorkerPoolHash`).
With the current approach of adding a value to the hashed data depending on the node local DNS flag, the option was to roll nodes of clusters having it enabled or disabled. I decided the former was likely the smaller number as the feature is officially still alpha while the latter includes probably most productive clusters.

Any ideas for improvement are welcome :-) .

Reply:
Maybe only consider the `spec.systemComponents` here instead of using the annotation to prevent unintended nodes roll out for existing shoots?

Reply:
Addressed with latest commit.